streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -174,7 +197,77 @@
+    }
+
+
+    /**
+     * Create a {@link KStream} from the specified topics.
+     * <p>
+     * If multiple topics are specified there are no ordering guaranteed for records from different topics.
+     * <p>
+     * Note that the specified input topics must be partitioned by key.
+     * If this is not the case it is the user's responsibility to repartition the date before any key based operation
+     * (like aggregation or join) is applied to the returned {@link KStream}.
+     *
+     * The source will use specified {@link TimestampExtractor} for extracting timestamps from records.
+     *
+     * @param timestampExtractor the {@link TimestampExtractor} used when consuming records; may be null if the source
+     * should use the {@link org.apache.kafka.streams.StreamsConfig#TIMESTAMP_EXTRACTOR_CLASS_CONFIG}, default timestampExtractor specified in the
+     * {@link org.apache.kafka.streams.StreamsConfig} stream configuration.
+     *                    offsets are available
+     * @param keySerde    key serde used to read this source {@link KStream},
+     *                    if not specified the default serde defined in the configs will be used
+     * @param valSerde    value serde used to read this source {@link KStream},
+     *                    if not specified the default serde defined in the configs will be used
+     * @param topics      the topic names; must contain at least one topic name
+     * @return a {@link KStream} for the specified topics
+     */
+    public <K, V> KStream<K, V> stream(final TimestampExtractor timestampExtractor,
+                                       final Serde<K> keySerde,
+                                       final Serde<V> valSerde,
+                                       final String... topics) {
+        final String name = newName(KStreamImpl.SOURCE_NAME);
+
+        addSource(null, name, timestampExtractor, keySerde == null ? null : keySerde.deserializer(), valSerde == null ? null : valSerde.deserializer(), topics);
+
+        return new KStreamImpl<>(this, name, Collections.singleton(name), false);
+    }
+
+
+    /**
+     * Create a {@link KStream} from the specified topics.
+     * <p>
+     * If multiple topics are specified there are no ordering guaranteed for records from different topics.
+     * <p>
+     * Note that the specified input topics must be partitioned by key.
+     * If this is not the case it is the user's responsibility to repartition the date before any key based operation
+     * (like aggregation or join) is applied to the returned {@link KStream}.
+     *
+     * The source will use specified {@link TimestampExtractor} for extracting timestamps from records.
+     *
+     * @param offsetReset the {@code "auto.offset.reset"} policy to use for the specified topics if no valid committed
+     *                    offsets are available
+     * @param timestampExtractor the {@link TimestampExtractor} used when consuming records; may be null if the source
```

Comment:
Nit: adjust indention of other parameters; line should not be longer than 120 chars. Indent second/third/etc line, too.
Text can be shorter:
```
the timestamp extractor used for this source {@link KStream}
if not specified the default extractor defined in the configs will be used
```
(no need to link to `TimestampExtractor` as there will be a link in the JavaDocs anyway.